### PR TITLE
[#7864] [Platform]Delete associated backups after deleting the universe.

### PR DESCRIPTION
**Pull request by mahendranbhat**
**Description**:
Currently, we are not deleting backups after the universe got delete. Give an option to the user whether to delete the backups or not while deleting the universe.

**Test Plan**: 

1. Created a universe and created some tables.
2. Taken a couple of backups.
3. Delete the universe by selecting delete backups in the modal. ( by default it will be turned off ) 
4. When the destroy universe finishes, the associated backups also deleted from the respective backup config.
5. If the option is not chosen then the backup should stay. Currently, we can not see it anywhere in the UI.
6. Scheduled backup's delete also tested with schedule with expiry time. (NFS)
7. Tested without deleting backup.

Screenshot from 2021-04-15 21-23-56

```[info] Test com.yugabyte.yw.commissioner.TaskGarbageCollectorTest.testPurge_invalidData started
[info] Test run finished: 0 failed, 0 ignored, 7 total, 0.242s
[error] Failed: Total 1310, Failed 1, Errors 0, Passed 1305, Skipped 4
[error] Failed tests:
[error]         com.yugabyte.yw.commissioner.tasks.MultiTableBackupTest
[error] (Test / test) sbt.TestsFailedException: Tests unsuccessful
[error] Total time: 1579 s (26:19), completed 12 May, 2021 11:47:30 PM
mahendra@mahendra-HP-ZBook-14u-G5:~/yugabyte/yugabyte-db/managed$ sbt "testOnly com.yugabyte.yw.commissioner.tasks.MultiTableBackupTest"
[info] welcome to sbt 1.3.13 (Private Build Java 1.8.0_292)
[info] loading settings for project managed-build from plugins.sbt ...
[info] loading project definition from /home/mahendra/yugabyte/yugabyte-db/managed/project
[info] loading settings for project root from ui-build.sbt,build.sbt ...
[info] set current project to yugaware (in build file:/home/mahendra/yugabyte/yugabyte-db/managed/)
[Yugabyte sbt log] [Resolver] Maven cache server (such as Nexus or Artifactory), specified by YB_MVN_CACHE_URL: List()
[Yugabyte sbt log] [Resolver] Local resolver (enabled by USE_MAVEN_LOCAL, path can be customized with YB_MVN_LOCAL_REPO): List(cache:Maven2 Local: /home/mahendra/.m2/repository)
[Yugabyte sbt log] [Resolver] Default resolver: Vector(FileRepository(local, Patterns(ivyPatterns=Vector(${ivy.home}/local/[organisation]/[module]/(scala_[scalaVersion]/)(sbt_[sbtVersion]/)([branch]/)[revision]/[type]s/[artifact](-[classifier]).[ext]), artifactPatterns=Vector(${ivy.home}/local/[organisation]/[module]/(scala_[scalaVersion]/)(sbt_[sbtVersion]/)([branch]/)[revision]/[type]s/[artifact](-[classifier]).[ext]), isMavenCompatible=false, descriptorOptional=false, skipConsistencyCheck=false), FileConfiguration(true, None)), public: https://repo1.maven.org/maven2/)
[Yugabyte sbt log] [Resolver] Snapshot resolver for yb-client jar (used when USE_MAVEN_LOCAL is not set, mostly during local development, configured with YB_MVN_SNAPSHOT_URL): List()
[info] Instrumenting 709 classes to /home/mahendra/yugabyte/yugabyte-db/managed/target/scala-2.12/jacoco/instrumented-classes
[info] Test run started
[info] Test com.yugabyte.yw.commissioner.tasks.MultiTableBackupTest.testTransactionalMultiTableBackupList started
[info] Test com.yugabyte.yw.commissioner.tasks.MultiTableBackupTest.testMultiTableBackupKeyspace started
[info] Test com.yugabyte.yw.commissioner.tasks.MultiTableBackupTest.testMultiTableBackup started
[info] Test com.yugabyte.yw.commissioner.tasks.MultiTableBackupTest.testYSQLBackupTables started
[info] Test com.yugabyte.yw.commissioner.tasks.MultiTableBackupTest.testTransactionalUniverseBackup started
[info] Test com.yugabyte.yw.commissioner.tasks.MultiTableBackupTest.testMultiTableBackupList started
[info] Test com.yugabyte.yw.commissioner.tasks.MultiTableBackupTest.testMultiTableBackupIgnore started
[info] Test run finished: 0 failed, 0 ignored, 7 total, 31.308s
[info] Passed: Total 7, Failed 0, Errors 0, Passed 7
[success] Total time: 37 s, completed 12 May, 2021 11:51:09 PM
```

### CODE REVIEW

**Comment by nishantSharma459** [comment]
DeleteBackups

File: managed/src/main/java/com/yugabyte/yw/commissioner/tasks/DestroyUniverse.java [hunk 1]
```diff
@@ -55,6 +66,53 @@ public void run() {
         universe = lockUniverseForUpdate(-1 , true);
       }
 
+      if (params().isDeleteBackups) {
+        List<Backup> backupList = Backup.fetchByUniverseUUID(params().customerUUID, universe.universeUUID);
+        for (Backup backup : backupList) {
+          try {
```
Comment (mahendranbhat) [inline managed/src/main/java/com/yugabyte/yw/commissioner/tasks/DestroyUniverse.java:72]: This part of the code is almost same as the delete backup task. Let me know where I can move this and use the common function in both files.

File: managed/src/main/java/com/yugabyte/yw/commissioner/tasks/DestroyUniverse.java [hunk 2]
```diff
@@ -55,6 +66,53 @@ public void run() {
...
+            backup = Backup.get(params().customerUUID, backup.backupUUID);
+            if (backup.state != Backup.BackupState.Completed) {
+              LOG.error("Cannot delete backup in any other state other than completed.");
```
Comment (SergeyPotachev) [inline managed/src/main/java/com/yugabyte/yw/commissioner/tasks/DestroyUniverse.java:75]: Should we insert `continue` here / after the log?

File: managed/src/main/java/com/yugabyte/yw/commissioner/tasks/DestroyUniverse.java [hunk 3]
```diff
@@ -55,6 +66,53 @@ public void run() {
...
+            if (backupParams.backupList != null) {
+              for (BackupTableParams childBackupParams : backupParams.backupList) {
+                childBackupParams.actionType = BackupTableParams.ActionType.DELETE;
+                ShellResponse response = tableManager.deleteBackup(childBackupParams);
+                JsonNode jsonNode = Json.parse(response.message);
+                if (response.code != 0 || jsonNode.has("error")) {
+                  // Revert state to completed since it couldn't get deleted.
+                  backup.transitionState(Backup.BackupState.Completed);
+                  LOG.error("Delete Backup failed for {}. Response code={}, hasError={}.",
+                            childBackupParams.storageLocation, response.code, jsonNode.has("error"));
```
Comment (SergeyPotachev) [inline managed/src/main/java/com/yugabyte/yw/commissioner/tasks/DestroyUniverse.java:88]: The same here - do we need `break` here? Because you are updating state of the whole backup...

File: managed/src/main/java/com/yugabyte/yw/commissioner/tasks/DestroyUniverse.java [hunk 4]
```diff
@@ -45,6 +55,7 @@ public void run() {
     try {
       // Create the task list sequence.
       subTaskGroupQueue = new SubTaskGroupQueue(userTaskUUID);
+      tableManager = Play.current().injector().instanceOf(TableManager.class);
```
Comment (SergeyPotachev) [inline managed/src/main/java/com/yugabyte/yw/commissioner/tasks/DestroyUniverse.java:58]: Btw, DeleteBackup initializes tableManager in the `initialize` function. I don't know enough pros and cons of such approach. Maybe @Arnav15 can suggest. But anyway, if you leave it here you don't need for a class field.

File: managed/src/main/java/com/yugabyte/yw/controllers/UniverseController.java [hunk 5]
```diff
@@ -1159,6 +1160,11 @@ public Result destroy(UUID customerUUID, UUID universeUUID) {
     if (request().getQueryString("isForceDelete") != null) {
       isForceDelete = Boolean.parseBoolean(request().getQueryString("isForceDelete"));
     }
+    boolean isDeleteBackups = false;
+    if (request().getQueryString("isForceDelete") != null) {
```
Comment (SergeyPotachev) [inline managed/src/main/java/com/yugabyte/yw/controllers/UniverseController.java:1164]: `isDeleteBackups` here instead of second `isForceDelete`.

**Review by SergeyPotachev — CHANGES_REQUESTED** [review]
Very weak tests coverage. + Some comments.

File: managed/ui/src/actions/universe.js [hunk 6]
```diff
@@ -210,9 +210,12 @@ export function resetUniverseList() {
   };
 }
 
-export function deleteUniverse(universeUUID, isForceDelete) {
+export function deleteUniverse(universeUUID, isForceDelete, isDeleteBackups) {
   const customerUUID = localStorage.getItem('customerId');
-  const deleteRequestPayload = { isForceDelete: isForceDelete };
+  const deleteRequestPayload = {
```
Comment (mjoshi0923) [inline managed/ui/src/actions/universe.js:215]:
```suggestion
  const deleteRequestPayload = {
      isForceDelete,
      isDeleteBackups,
    };
```
No need of colon notation you can change it to above code.

File: managed/ui/src/components/universes/DeleteUniverse/DeleteUniverse.js [hunk 7]
```diff
@@ -62,7 +67,7 @@ export default class DeleteUniverse extends Component {
     } = this.props;
     this.props.onHide();
     if (type === 'primary') {
-      this.props.submitDeleteUniverse(data.universeUUID, this.state.isForceDelete);
+      this.props.submitDeleteUniverse(data.universeUUID, this.state.isForceDelete, this.state.isDeleteBackups);
```
Comment (mjoshi0923) [inline managed/ui/src/components/universes/DeleteUniverse/DeleteUniverse.js:70]: Destruct submitDeleteUniverse from props and then use it.

File: managed/ui/src/components/universes/DeleteUniverse/DeleteUniverse.js [hunk 8]
```diff
@@ -104,11 +109,18 @@ export default class DeleteUniverse extends Component {
...
         footerAccessory={
-          <YBCheckBox
-            label={'Ignore Errors and Force Delete'}
-            className="footer-accessory"
-            input={{ checked: this.state.isForceDelete, onChange: this.toggleForceDelete }}
-          />
+          <div className="force-delete">
+            <YBCheckBox
+              label={'Ignore Errors and Force Delete'}
+              className="footer-accessory"
```
Comment (mjoshi0923) [inline managed/ui/src/components/universes/DeleteUniverse/DeleteUniverse.js:115]: label="Ignore Errors and Force Delete"

File: managed/src/main/java/com/yugabyte/yw/commissioner/tasks/DestroyUniverse.java (hunk 2 above)
Comment (mahendranbhat) [reply managed/src/main/java/com/yugabyte/yw/commissioner/tasks/DestroyUniverse.java:75]: yeah, we should.

File: managed/src/main/java/com/yugabyte/yw/commissioner/tasks/DestroyUniverse.java (hunk 3 above)
Comment (mahendranbhat) [reply managed/src/main/java/com/yugabyte/yw/commissioner/tasks/DestroyUniverse.java:89]: yeah, but I am not sure what this part of code doing. @Arnav15 can you please help me to understand this?
Is this related to backing up multiple tables? ( entire universe )

File: managed/src/main/java/com/yugabyte/yw/commissioner/tasks/DestroyUniverse.java (hunk 1 above)
Comment (mahendranbhat) [reply managed/src/main/java/com/yugabyte/yw/commissioner/tasks/DestroyUniverse.java:72]: @Arnav15

File: managed/src/main/java/com/yugabyte/yw/commissioner/tasks/DestroyUniverse.java (hunk 2 above)
Comment (mahendranbhat) [reply managed/src/main/java/com/yugabyte/yw/commissioner/tasks/DestroyUniverse.java:74]: Should we raise an exception once we get a backup that is still in progress while deleting the universe? ( guess universe lock should take care of it but still )

File: managed/ui/src/components/universes/DeleteUniverse/DeleteUniverse.js (hunk 8 above)
Comment (nishantSharma459) [reply managed/ui/src/components/universes/DeleteUniverse/DeleteUniverse.js:117]: Done.

File: managed/ui/src/components/universes/DeleteUniverse/DeleteUniverse.js (hunk 7 above)
Comment (nishantSharma459) [reply managed/ui/src/components/universes/DeleteUniverse/DeleteUniverse.js]: Done

File: managed/ui/src/actions/universe.js (hunk 6 above)
Comment (nishantSharma459) [reply managed/ui/src/actions/universe.js:215]: Done.

File: managed/src/main/java/com/yugabyte/yw/commissioner/tasks/DestroyUniverse.java (hunk 1 above)
Comment (Arnav15) [reply managed/src/main/java/com/yugabyte/yw/commissioner/tasks/DestroyUniverse.java:71]: Hmm, @mahendranbhat do you think we can make this a subtask? That way we can use the same subtask in both tasks. Let me know if that makes sense?

File: managed/src/main/java/com/yugabyte/yw/commissioner/tasks/DestroyUniverse.java (hunk 2 above)
Comment (Arnav15) [reply managed/src/main/java/com/yugabyte/yw/commissioner/tasks/DestroyUniverse.java:74]: Nope, in the case of a force delete, this can be a legitimate issue, coz in that case we don't take care of the locks. Also, for backups, the universe isn't actually locked while the actual bulk of the work is happening. Maybe we can add a check to see if there are backups in progress, and if so, we can just a) either wait it out, b) just fail that subtask and try deleting it anyway.

File: managed/src/main/java/com/yugabyte/yw/commissioner/tasks/DestroyUniverse.java (hunk 3 above)
Comment (Arnav15) [reply managed/src/main/java/com/yugabyte/yw/commissioner/tasks/DestroyUniverse.java:88]: Yes, this is if there are multiple backup objects as part of a single backup (Universe, keyspace, namespace, etc., multi-table)

**Review by Arnav15 — CHANGES_REQUESTED** [review]

File: managed/src/main/java/com/yugabyte/yw/commissioner/tasks/DestroyUniverse.java (hunk 1 above)
Comment (mahendranbhat) [reply managed/src/main/java/com/yugabyte/yw/commissioner/tasks/DestroyUniverse.java:71]: we have to raise an exception in the backup task but here we should not if delete backup failed for any reason ( I assume, correct me if I am wrong ). So in the subtask, one more param to identify whether the subtask can raise an exception or not? Or can we keep it like this only?

File: managed/src/main/java/com/yugabyte/yw/commissioner/tasks/DestroyUniverse.java (hunk 2 above)
Comment (mahendranbhat) [reply managed/src/main/java/com/yugabyte/yw/commissioner/tasks/DestroyUniverse.java:74]: @Arnav15 waiting for the backup to complete or fail ( other than in progress ) before moving to the next code blocks makes sense to me. LMK if I can go ahead with the change.

**Review by sshev — COMMENTED** [review]
@nishantSharma459 having a second inlined checkbox in the modal footer looks odd to me.
I'd rather expected it right below the text input. Is this an approved UI?

File: managed/src/main/java/com/yugabyte/yw/commissioner/tasks/DestroyUniverse.java (hunk 1 above)
Comment (Arnav15) [reply managed/src/main/java/com/yugabyte/yw/commissioner/tasks/DestroyUniverse.java:71]: Yes @mahendranbhat , that is correct. Though, I think a force delete task ignores subtask failures anyway, so you might not have to build it in the subtask. The actual task framework should take care of it. So basically you will create a new subtask, and you can actually put them in the same subtask group and run all of them together. The task framework already has a mechanism to ignore errors if one subtask in the group fails.

**Comment by nishantSharma459** [comment]
@sshev There was a doc shared across which straight the alignment of the `new checkbox` I followed that.

Doc link: https://docs.google.com/document/d/1vPxVvBD2QNVg_fxswfve2-Tr4RUWagii_yoJgwLYGQI/edit#heading=h.itboeyc7fqfc

and I've tried to put it underneath but it wasn't looking good. There was an empty space under action buttons.

**Review by sshev — APPROVED** [review]
JS code looks good, pls check with BE folks on the other changes

File: managed/src/main/java/com/yugabyte/yw/commissioner/tasks/DestroyUniverse.java (hunk 2 above)
Comment (mahendranbhat) [reply managed/src/main/java/com/yugabyte/yw/commissioner/tasks/DestroyUniverse.java]: Screenshot from 2021-04-15 21-54-55
If we make it a subtask , this page looks like this.. Is that fine? @SergeyPotachev @Arnav15

File: managed/src/main/java/com/yugabyte/yw/commissioner/tasks/UniverseTaskBase.java [hunk 9]
```diff
@@ -999,6 +999,20 @@ public SubTaskGroup createTableBackupTask(BackupTableParams taskParams, Backup b
     return subTaskGroup;
   }
 
+   public SubTaskGroup createDeleteBackupTasks(List<Backup> backups, UUID customerUUID) {
```
Comment (sb-yb) [inline managed/src/main/java/com/yugabyte/yw/commissioner/tasks/UniverseTaskBase.java:1002]: Perfect! 
I was looking at previous commit and was going to comment:
"Have not looked deeply but can we just submit DeleteBackup task instead of reinventing the wheel?
Or at least share code with that task. But I prefer former. "

File: managed/src/test/java/com/yugabyte/yw/commissioner/tasks/DestroyUniverseTest.java [hunk 10]
```diff
@@ -82,6 +88,40 @@ public void testReleaseUniverseAndResolveAlerts() {
...
+    
+    DestroyUniverse.Params taskParams = new DestroyUniverse.Params();
+    taskParams.universeUUID = defaultUniverse.universeUUID;
+    taskParams.customerUUID = defaultCustomer.uuid;
+    taskParams.isForceDelete = Boolean.FALSE;
+    taskParams.isDeleteBackups = Boolean.TRUE;
+
+    submitTask(taskParams, 4);
+    assertFalse(Universe.checkIfUniverseExists(defaultUniverse.name));
+  }
```
Comment (SergeyPotachev) [inline managed/src/test/java/com/yugabyte/yw/commissioner/tasks/DestroyUniverseTest.java:106]: Should we check that the backup is removed as well?

File: managed/src/test/java/com/yugabyte/yw/commissioner/tasks/DestroyUniverseTest.java [hunk 11]
```diff
@@ -82,6 +88,40 @@ public void testReleaseUniverseAndResolveAlerts() {
...
+    b.transitionState(Backup.BackupState.Completed);
+    
+    DestroyUniverse.Params taskParams = new DestroyUniverse.Params();
+    taskParams.universeUUID = defaultUniverse.universeUUID;
+    taskParams.customerUUID = defaultCustomer.uuid;
+    taskParams.isForceDelete = Boolean.FALSE;
+    taskParams.isDeleteBackups = Boolean.FALSE;
+
+    submitTask(taskParams, 4);
+    assertFalse(Universe.checkIfUniverseExists(defaultUniverse.name));
```
Comment (SergeyPotachev) [inline managed/src/test/java/com/yugabyte/yw/commissioner/tasks/DestroyUniverseTest.java:122]: And here we need to check that the backup wasn't removed.

File: managed/src/test/java/com/yugabyte/yw/controllers/UniverseControllerTest.java [hunk 12]
```diff
@@ -930,6 +935,44 @@ public void run(Universe universe) {
...
+        UserIntent userIntent = new UserIntent();
+        userIntent.providerType = CloudType.aws;
+        universeDetails.upsertPrimaryCluster(userIntent, null);
+        universe.setUniverseDetails(universeDetails);
+      }
+    };
+    // Save the updates to the universe.
+    Universe.saveDetails(u.universeUUID, updater);
+
+    Backup b = ModelFactory.createBackup(customer.uuid,
```
Comment (SergeyPotachev) [inline managed/src/test/java/com/yugabyte/yw/controllers/UniverseControllerTest.java:959]: Reformat the line.

File: managed/src/test/java/com/yugabyte/yw/controllers/UniverseControllerTest.java [hunk 13]
```diff
@@ -930,6 +935,44 @@ public void run(Universe universe) {
...
+    b.transitionState(Backup.BackupState.Completed);
+
+    String url = "/api/customers/" + customer.uuid + "/universes/"
+        + u.universeUUID + "?isForceDelete=true&isDeleteBackups=true";
+    Result result = doRequestWithAuthToken("DELETE", url, authToken);
+    assertOk(result);
+    JsonNode json = Json.parse(contentAsString(result));
+    assertValue(json, "taskUUID", fakeTaskUUID.toString());
+
+    CustomerTask th = CustomerTask.find.query().where().eq("task_uuid", fakeTaskUUID).findOne();
```
Comment (SergeyPotachev) [inline managed/src/test/java/com/yugabyte/yw/controllers/UniverseControllerTest.java:970]: Strange name for the CustomerTask instance.

**Review by SergeyPotachev — CHANGES_REQUESTED** [review]
Only a few small comments for the BE part.

File: managed/src/main/java/com/yugabyte/yw/commissioner/tasks/DestroyUniverse.java (hunk 2 above)
Comment (mahendranbhat) [reply managed/src/main/java/com/yugabyte/yw/commissioner/tasks/DestroyUniverse.java]: @sb-yb

**Comment by sb-yb** [comment]
pl merge conflicting files before testing

File: managed/src/main/java/com/yugabyte/yw/commissioner/tasks/DestroyUniverse.java (hunk 2 above)
Comment (sb-yb) [reply managed/src/main/java/com/yugabyte/yw/commissioner/tasks/DestroyUniverse.java]: I think its because it does not have subtasks and percentage is calculated as num_completed/num_total subtasks. 
May be if num_total  is 0 we should set percentage to 100 instead of 0. WDYT @Arnav15 @SergeyPotachev ?

Comment (SergeyPotachev) [reply managed/src/main/java/com/yugabyte/yw/commissioner/tasks/DestroyUniverse.java]: @sb-yb Yes, makes sense.

File: managed/src/main/java/com/yugabyte/yw/commissioner/tasks/DestroyUniverse.java [hunk 14]
```diff
@@ -21,11 +24,16 @@
 import com.yugabyte.yw.commissioner.tasks.subtasks.RemoveUniverseEntry;
 import com.yugabyte.yw.common.AlertManager;
 import com.yugabyte.yw.common.DnsManager;
+import com.yugabyte.yw.common.ShellResponse;
```
Comment (SergeyPotachev) [inline managed/src/main/java/com/yugabyte/yw/commissioner/tasks/DestroyUniverse.java:27]: Please check for unused imports. It looks like some of them are not used.

File: managed/src/test/java/com/yugabyte/yw/controllers/UniverseControllerTest.java [hunk 15]
```diff
@@ -930,6 +935,45 @@ public void run(Universe universe) {
     assertAuditEntry(1, customer.uuid);
   }
 
+  @Test
+  public void testUniverseDestroyValidUUIDIsForceDeleteAndDeleteBackup() {
```
Comment (SergeyPotachev) [inline managed/src/test/java/com/yugabyte/yw/controllers/UniverseControllerTest.java:939]: Maybe to make this test parametrised and to check with isDeleteBackups=true, isDeleteBackups=false and isDeleteBackups is absent.

**Review by SergeyPotachev — CHANGES_REQUESTED** [review]
Only two small comments.

**Review by SergeyPotachev — APPROVED** [review]
Looks good to me!

File: managed/src/main/java/com/yugabyte/yw/commissioner/tasks/DestroyUniverse.java (hunk 2 above)
Comment (mahendranbhat) [reply managed/src/main/java/com/yugabyte/yw/commissioner/tasks/DestroyUniverse.java]: `public double getPercentCompleted() {
    int numSubtasks = TaskInfo.find.query().where()
      .eq("parent_uuid", getTaskUUID())
      .findCount();
    if (numSubtasks == 0) {
      return 0.0;
    }
    int numSubtasksCompleted = TaskInfo.find.query().where()
      .eq("parent_uuid", getTaskUUID())
      .eq("task_state", TaskInfo.State.Success)
      .findCount();
    return numSubtasksCompleted * 100.0 / numSubtasks;
  }`
  yes , but do we have to return 100 if no of the subtask is 0? ( does not make sense if it gets failed.. but even 0 not making sense when the task is completed successfully.

Comment (mahendranbhat) [reply managed/src/main/java/com/yugabyte/yw/commissioner/tasks/DestroyUniverse.java]: yes , but do we have to return 100 if no of the subtask is 0? ( does not make sense if it gets failed.. but even 0 not making sense when the task is completed successfully.

File: managed/src/main/java/com/yugabyte/yw/models/TaskInfo.java [hunk 16]
```diff
@@ -273,7 +273,7 @@ public double getPercentCompleted() {
       .eq("parent_uuid", getTaskUUID())
       .findCount();
     if (numSubtasks == 0) {
-      return 0.0;
+      return 100.0;
```
Comment (jaydeepkumara) [inline managed/src/main/java/com/yugabyte/yw/models/TaskInfo.java:276]: why this change?

Comment (mahendranbhat) [reply managed/src/main/java/com/yugabyte/yw/models/TaskInfo.java:276]: @jaydeepkumara https://github.com/yugabyte/yugabyte-db/pull/7864#discussion_r622530835

File: managed/ui/src/components/universes/DeleteUniverse/DeleteUniverse.scss [hunk 17]
```diff
@@ -0,0 +1,5 @@
+/* Copyright (c) YugaByte, Inc. */
+
+.paused-note {
+    color: #e8473f;
+}
```
[the remark this answers is not in the document]
Comment (nishantSharma459) [reply managed/ui/src/components/universes/DeleteUniverse/DeleteUniverse.scss]: The file has been removed.

File: managed/ui/src/components/universes/DeleteUniverse/DeleteUniverse.js [hunk 18]
```diff
@@ -34,13 +40,29 @@ export default class DeleteUniverse extends Component {
...
     return (
-      <div>
-        {body}
+      <>
+        {universePaused ?
+          <>
+            Are you sure you want to delete the universe?
+            <br />
+            <br />
+            <span className='paused-note'>
```
[the remark this answers is not in the document]
Comment (nishantSharma459) [reply managed/ui/src/components/universes/DeleteUniverse/DeleteUniverse.js]: Done.

**Comment by nishantSharma459** [comment]
Here's a screenshot with the Alert box.

Screenshot from 2021-05-12 14-17-01

**Review by sshev — APPROVED** [review]
Frontend looks good to me
